### ISSUE TITLE
Retroarch compiled with only wayland support doesn't launch

### ISSUE THREAD

**Issue by iam-angel-val** [NixOS/nixpkgs#349905]
### Describe the bug
Retroarch immediately crashes when built without x11 support and with wayland support. This sucks as working wayland support would let HDR work as well, allowing users to use things like CRT megatron shaders.

### Steps To Reproduce
1. Add this to your config:
```nix
environment.systemPackages = [
  (retroarch.override {
    retroarch = (retroarchBare.overrideAttrs
      (final: prev: {
        configureFlags = prev.configureFlags ++ ["--enable-wayland" "--disable-x11"];
      })
    )
    .override {withWayland = true;};
    cores = with libretro; [
      # etc
    ];
  })
];
```
2. Launch retroarch, with the `-v` flag for logs

### Expected behavior
Immediately closes, leaving these logs:
```
[INFO] [Input]: Found input driver: "sdl2".
[ERROR] Couldn't find any cloud sync driver named ""
[INFO] Available cloud sync drivers are:
[INFO] 	webdav
[INFO] 	null
[WARN] Going to default to first cloud sync driver...
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.01, Sample rate: 48000.00 Hz.
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "khr_display".
[INFO] [Vulkan]: Detecting screen resolution: 0x0.
[INFO] [Vulkan]: Found GPU at index 0: "<gpu info>".
[INFO] [Vulkan]: Using GPU index 0.
[ERROR] [Vulkan]: Failed to create KHR_display surface.
[ERROR] [Vulkan]: Failed to set video mode.
[ERROR] [Video]: Cannot open video driver.. Exiting..
[ERROR] Fatal error received in: "video_driver_init_internal()"
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
```

### Additional context
My suspicion of why this is happening is because retroarch cannot find a wayland context to attach to (or whatever more technical terms describe the situation better), and then crashes because it's the only thing I built it with support for. But it's difficult to tell, because retroarch has zero debug logging for when no display context is found (see [this function](https://github.com/libretro/RetroArch/blob/19b89463727f3334be09304e0dd8d13e78e23367/gfx/common/vulkan_common.c#L1531) in the retroarch source code)

### Notify maintainers

@matthewbauer @kolbycrouch

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.57, NixOS, 24.11 (Vicuna), 24.11.20241018.4c2fcb0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/riqkpszjqk02bi1wppfg8ip5xvh102qd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Frontear** [comment]
Do you know if it's an upstream issue or a nixpkgs-specific issue?

**Comment by iam-angel-val** [comment]
I'm decently sure that someone would notice if retroarch was completely unable to run under wayland, but I will admit I haven't tried it on other distros or windows. I have however tried the nixpkgs stable versions of the packages and the issue persisted.

**Comment by iam-angel-val** [comment]
I figured it out! Just add wayland-scanner to `retroarch-bare`'s `buildInputs` and it launches in wayland no problem!

for anyone wondering how to get HDR working, you should look at [this issue](https://github.com/NixOS/nixpkgs/pull/345805/files#diff-ce83891e02effc12d8975be1cbfe438bfaebf8259f057b3935c5539241cdc85b), and once you've got the package in your config, launch retroarch with the `ENABLE_HDR_WSI=1` envvar. Hope this helped someone, it's great to finally get this working after a couple weeks of trial and error :)

**Comment by iam-angel-val** [comment]
Fixed with #351597
